internal/generate/compile.go

```diff
@@ -205,3 +229,21 @@ func (c *compiler) sortDescriptors() {
 		}
 	}
 }
+
+func (c *compiler) skipCompile(id dbc.MessageID) bool {
+	if id == dbc.IndependentSignalsMessageID {
+		return true
+	}
+
+	if c.onlyCompile != nil {
+		for _, allowedMessageId := range c.onlyCompile {
+			if allowedMessageId == id {
+				return false
+			}
+		}
+
+		return true
+	}
+
+	return false
```

Comment:
Since it is completely safe to range over nil-slices (the body won't be called, as can be seen here: https://go.dev/play/p/EsOj94_q05_p) consider removing the extraneous nil check: 

```suggestion
  for _, allowedMessageId := range c.onlyCompile {
	  if allowedMessageId == id {
	  	return false
	  }
		return true
	}
```

Moreover, I think that `c.onlyCompileIds` is clearer about the intent in the code.